engine/runtime/src/test/scala/org/enso/compiler/test/codegen/AstToIrTest.scala

```diff
@@ -246,6 +246,48 @@ class AstToIrTest extends CompilerTest with Inside {
       consPat.fields(1) shouldBe an[IR.Pattern.Name]
     }
 
+    "support type patterns " in {
+      val ir =
+        """
+          |case foo of
+          |    f : Foo -> process_foo f
+          |    b : Bar -> process_bar b
+          |""".stripMargin.toIrExpression.get.asInstanceOf[IR.Case.Expr]
+
+      ir.branches.head.pattern shouldBe an[IR.Pattern.Type]
+
+      val tpePattern1 =
+        ir.branches(0).pattern.asInstanceOf[IR.Pattern.Type]
+
+      tpePattern1.name.name shouldEqual "f"
+      tpePattern1.tpe.name shouldEqual "Foo"
+
+      val tpePattern2 =
+        ir.branches(1).pattern.asInstanceOf[IR.Pattern.Type]
+
+      tpePattern2.name.name shouldEqual "b"
+      tpePattern2.tpe.name shouldEqual "Bar"
+    }
+
+    "support type patterns nested in constructor pattern " in {
+      val ir =
+        """
+          |case foo of
+          |    Cons (f : Foo) b -> process_foo f b
```

Comment:
I haven't seen this style yet in the `.enso` code above.

Reply:
It's because parser would reject it :)